### DISCUSSION TITLE
Option to disable preview in the builder

### DISCUSSION THREAD

**Discussion by R2bEEaton** [Budibase/budibase#7443] [Ideas]
I have some complicated pages on my app, and a couple of them require data sources with high limits. In reality though, a normal user would enter the page and the source would be filtered down to a manageable limit.

However, in the builder, the filters don't work because the URL variable isn't set, for example. In this case, the data provider returns all of the data which freezes the client and makes editing nigh on impossible sometimes.

I found four solutions, none of which are ideal:
1. Making the change within 2-3 seconds before the builder loads the preview.
2. Change the data source to be limited (which you have to do within 2-3 seconds before the builder loads the preview).
3. Wait minutes for the page to fully load and become responsive.
4. Not make the edit you want because it's a pain to deal with.

My proposed solution is a global toggle somewhere in the builder where you can disable the site preview. _Another option would be to be able to set URL variables within the builder, for example, or a default value for data source filters. This might be preferable for UX purposes._

**Comment by melohagan** [comment]
Hey @R2bEEaton 

This is a good point. I've seen others complain about this and have come across it myself. 

Personally I think adding a limit cap within the builder might be sufficient. I've raised the issue here anyway: https://github.com/Budibase/budibase/issues/7502
maybe some of the other developers will have some ideas too!
